test/Datadog.Trace.Tests/Agent/Thrift/ThriftUdpClientTransportTests.cs

```diff
@@ -0,0 +1,155 @@
+#pragma warning disable SA1636 // File header copyright text must match
+// <copyright file="ThriftUdpClientTransportTests.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+#pragma warning restore SA1636 // File header copyright text must match
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Datadog.Trace.Agent.Jaeger;
+using Moq;
+using Thrift.Transport;
+using Xunit;
+
+namespace Datadog.Trace.Tests.Agent.Thrift
+{
+    public class ThriftUdpClientTransportTests : IDisposable
+    {
+        private readonly Mock<IJaegerClient> mockClient = new Mock<IJaegerClient>();
+        private MemoryStream testingMemoryStream = new MemoryStream();
+
+        public void Dispose()
+        {
+            this.testingMemoryStream?.Dispose();
+        }
+
+        [Fact]
+        public void Constructor_ShouldConnectClient()
+        {
+            var host = "host, yo";
+            var port = 4528;
+
+            new JaegerThriftClientTransport(host, port, this.testingMemoryStream, this.mockClient.Object);
+
+            this.mockClient.Verify(t => t.Connect(host, port), Times.Once);
```

Comment:
Why a constructor should connect? 

**IMPORTANT TEST** 
What happens if the "connection" gets disconnected. For example, if stop the OTel collector (or Jager if you send test with it directly) and bring it up after a few minutes? I suggest to test it on Linux/MacOS and Windows the Sockets work differently on those OSes.

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/78#discussion_r595989157